tests/WallsComputationTest.cpp

```diff
@@ -149,12 +149,12 @@ TEST_F(WallsComputationTest, WallToolPathsGetWeakOrder)
     walls_computation.generateWalls(&layer);
 
     const bool outer_to_inner = false;
-    std::vector<const ExtrusionLine*> all_paths;
+    std::vector<ExtrusionLine> all_paths;
     for (auto& inset : part.wall_toolpaths)
     {
         for (auto& line : inset)
```

Comment:
Untested of course, but I think we could do something like this here. Not a required change, I just think it is cool.

```suggestion
    for (auto& line : part.wall_toolpaths | views::join)
    {
```

Since this is a vector\<vector\<ExtrusionLine>>

Or even better

```suggestion
    std::vector<const ExtrusionLine*> all_paths = part.wall_toolpaths | action::join;
```

Reply:
The return type of `actions::join` is a view not a `vector`. Use `auto` for complex types

Reply:
https://github.com/Ultimaker/CuraEngine/pull/1761/commits/2944616106d8e100504d35b3a6c18a843f18100e